_exception.py

```diff
@@ -0,0 +1,16 @@
+class DestinationUnreachableException(Exception):
+    def __init__(self, message="Destination was reported unreachable."):
+        super(Exception, self).__init__(message)
+
+
+class TimeToLiveExceededException(Exception):
+    def __init__(self, reporter_address, message=None):
+        self.reporter_address = reporter_address
+        m = message if message else \
+            "Address %s reported that the packet time to live was exceeded." % reporter_address
```

Comment:
> "Address %s reported that the packet time to live was exceeded." % reporter_address
> TypeError: not all arguments converted during string formatting

Actually, the addr is a `tuple(ip, port)`. So we can use `.format()` instead of `%` and use `reporter_address[0]` instead of `reporter_address`